File: evidence/generated-systems/angular/aerospace-on-angular/src/app/models/AerospaceManufacturer.js

// Define collection and schema for AerospaceManufacturer
export interface AerospaceManufacturer {
    name:
	type : string
    legalName:
	type : string
    headquartersCountry:
	type : string
    website:
	type : string
    Programs:
 	type : [{ type: Schema.Types.ObjectId, ref: 'AircraftProgram' }]
    Plants:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Plant' }]
    Suppliers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Supplier' }]
    ProductionCertificates:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ProductionCertificate' }]
#
    collection: 'aerospaceManufacturers'
}

File: evidence/generated-systems/django/aerospace-on-django/aerospaceOnDjango/models/AerospaceManufacturer.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model AerospaceManufacturer
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class AerospaceManufacturer Declaration
#======================================================================
class AerospaceManufacturer (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	legalName = models.CharField(max_length=200, null=True)
	headquartersCountry = models.CharField(max_length=200, null=True)
	website = models.CharField(max_length=200, null=True)
	programs = models.ManyToManyField('AircraftProgram',  blank=True, related_name='+')
	plants = models.ManyToManyField('Plant',  blank=True, related_name='+')
	suppliers = models.ManyToManyField('Supplier',  blank=True, related_name='+')
	productionCertificates = models.ManyToManyField('ProductionCertificate',  blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.legalName
		str = str + self.headquartersCountry
		str = str + self.website
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "AerospaceManufacturer";
    
	def objectType(self):
		return "AerospaceManufacturer";
